docs/source/daml2ts/index.rst

```diff
@@ -0,0 +1,290 @@
++-------------------+------------------+----------------------------------+
+| ``Bool``          | ``Bool``         | ``boolean``                      |
++-------------------+------------------+----------------------------------+
+| ``Int``           | ``Int``          | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Decimal``       | ``Decimal``      | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Numeric``       | ``Numeric n``    | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Text``          | ``Text``         | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Time``          | ``Time``         | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Party``         | ``Party``        | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``List<T>``       | ``[t]``          | ``T[]``                          |
++-------------------+------------------+----------------------------------+
+| ``Date``          | ``Date``         | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``ContractId<T>`` | ``ContractId t`` | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Optional<T>``   | ``Optional t``   | ``null | (null extends T ?``     |
+|                   |                  | ``[] | [Exclude<T, null>] : T)`` |
++-------------------+------------------+----------------------------------+
+| ``TextMap<T>``    | ``TextMap t``    | ``{ [key: string]: T }``         |
++-------------------+------------------+----------------------------------+
+
+.. note::
+   The TypeScript types ``Time``, ``Decimal``, ``Numeric`` and ``Int`` all alias to ``string``. These choices relate to the avoidance of precision loss under serialization over the `json-api <../json-api/index.html>`_.
+
+.. note::
+   The Typescript definition of type ``Optional<T>`` in the above table might look complicated. It accounts for differences in the encoding of optional values when nested versus when they are not (i.e. "top-level"). For example, ``null`` and ``"foo"`` are two possible values of ``Optional<Text>`` whereas, ``[]`` and ``["foo"]`` are two possible values of type ``Optional<Optional<Text>>`` (``null`` is another possible value, ``[null]`` is **not**).
+
+DAML to TypeScript mappings
+---------------------------
+
+The mappings from DAML to TypeScript are best explained by example.
+
+Records
+~~~~~~~
+
+In DAML, we might model a person like this.
+
+.. code-block:: daml
+   :linenos:
+
+   data Person =
+     Person with
```

Comment:
I would move this on the line before. Then the DAML and the TS look a bit closer.

Reply:
✔️